### ISSUE TITLE
Line of code output does not match the original file

### ISSUE THREAD

**Issue by luiscruz** [facebook/infer#317]
I am having the following output with my checker

```
k9mail/src/main/java/com/fsck/k9/notification/DeviceNotifications.java:221: error: CHECKERS_ANDROID_INTERNAL_ACCESSORS
  call to internal accessor method getReplyActionIcon
  219.       private void addReplyAction(Builder builder, NotificationContent content, int notificationId) {
  220.           int icon = getReplyActionIcon();
  221. >         String title = context.getString(R.string.notification_action_reply);
  222.
```

The line that should be highlighted in this output is `int icon = getReplyActionIcon();` which here is the 220.
When I check the original file, the line 221 is the line that has to be highlighted ( `int icon = getReplyActionIcon();`)

**Comment by jvillard** [comment]
Yeah, usually this is because the JVM bytecode sometimes has inaccurate information about the line number. I'm afraid there's not much we can do.

**Comment by luiscruz** [comment]
I don't think I've explained the issue properly.
The line code is correct but the output is not displaying it correctly.

Consider the following output:

```
app/src/main/java/net/bierbaumer/otp_authenticator/Entry.java:51: error: CHECKERS_ANDROID_INTERNAL_ACCESSORS
  call to internal accessor method setSecret
  49.       public Entry (JSONObject jsonObj ) throws JSONException {
  50.           this.setSecret(new Base32().decode(jsonObj.getString(JSON_SECRET)));
  51. >         this.setLabel(jsonObj.getString(JSON_LABEL));
  52.       }
  53.
```

According to this output I should be able to find a call to setSecret at line 51 of file ```app/src/main/java/net/bierbaumer/otp_authenticator/Entry.java```.

So let's check if the information is correct:
```pygmentize -O linenos=1  tmp/0xbb/otp-authenticator/app/src/main/java/net/bierbaumer/otp_authenticator/Entry.java | sed '51s/$/ <======= BUG HERE =======/'| sed -n '47,56p```
With this command I am asking to display code between lines 47 and 56 and append some message at line 51.

I get this output:

![screen shot 2016-03-23 at 15 24 50](https://cloud.githubusercontent.com/assets/1080714/13990214/69b9f918-f10b-11e5-9be9-68fffb3ee5a2.png)

As you can see the information was correct. It doesn't seem to be a problem of JVM byte code.

**Comment by jvillard** [comment]
Does this happen only with errors from your checker?

**Comment by luiscruz** [comment]
Good question... let me check!

**Comment by luiscruz** [comment]
I got the same issue with CHECKERS_FRAGMENT_RETAINS_VIEW:
```
k9mail/src/main/java/com/fsck/k9/fragment/MessageListFragment.java:841: error: CHECKERS_FRAGMENT_RETAINS_VIEW
  Fragment com.fsck.k9.fragment.MessageListFragment does not nullify View field mPullToRefreshView (type com.handmark.pulltorefresh.library.PullToRefreshListView) in onDestroyView. If this Fragment is placed on the back stack, a reference to this (probably dead) View will be retained. In general, it is a good idea to initialize View's in onCreateView, then nullify them in onDestroyView.
  839.       @Override
  840.       public void onDestroyView() {
  841. >         mSavedListState = mListView.onSaveInstanceState();
  842.           super.onDestroyView();
  843.

```

The line 841 should be ```public void onDestroyView() {```

**Comment by jvillard** [comment]
There was indeed an issue with wrong line numbers that have been fixed in master: bfb0cfa4fd65089094fd6f552e72689d642b9788 and 81856c0d15d14b823431db983152d4488779af08

It looks like you were hitting the same issue. Please re-open if not.
